hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcServer.java

```diff
@@ -108,28 +147,34 @@ public NettyRpcServer(Server server, String name, List<BlockingServiceAndInterfa
     if (config == null) {
       config = new NettyEventLoopGroupConfig(conf, "NettyRpcServer");
     }
+
+    // call before creating bootstrap below so that the necessary configs can be set
+    configureNettyWatermarks(conf);
+
     EventLoopGroup eventLoopGroup = config.group();
     Class<? extends ServerChannel> channelClass = config.serverChannelClass();
-    ServerBootstrap bootstrap = new ServerBootstrap().group(eventLoopGroup).channel(channelClass)
+    bootstrap = new ServerBootstrap().group(eventLoopGroup).channel(channelClass)
```

Comment:
Why we need to make this a class field?

Reply:
I originally did this because in my updateConfiguration code I was changing the bootstrap childOptions. But more recently I moved these updatable fields into the childHandler/initializer, so this is no longer necessary.

I'm working on adding tests right now, and have cleaned this up already. Will fix all your other comments as part of that, should hopefully have an update shortly.